### PR TITLE
Dynamic containers

### PR DESCRIPTION
**Pull request by bencameron**
Created IContainerFactory; Moved original logic for resolving blob from subpath into DefaultContainerFactory; Added constructor to AzureBlobFileProvider that accepts IContainerFactory;

### CODE REVIEW

File: src/Strathweb.AspNetCore.AzureBlobFileProvider/IContainerFactory.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Microsoft.WindowsAzure.Storage.Blob;
+
+namespace Strathweb.AspNetCore.AzureBlobFileProvider
+{
+    public interface IContainerFactory
```
Comment (filipw) [inline src/Strathweb.AspNetCore.AzureBlobFileProvider/IContainerFactory.cs]: sorry for the delay 😢
can we please rename it to `IBlobContainerFactory`? same for the default implementation

**Comment by filipw** [comment]
thanks a lot - looks good! I just left a small comment

File: src/Strathweb.AspNetCore.AzureBlobFileProvider/IContainerFactory.cs (hunk 1 above)
Comment (bencameron) [reply src/Strathweb.AspNetCore.AzureBlobFileProvider/IContainerFactory.cs]: No problem, and thanks for the feedback!  I've made the renames and pushed.

**Comment by filipw** [comment]
thanks!

**Pull request merged**